src/internal.c

```diff
@@ -11168,7 +11168,15 @@ int DoVerifyCallback(WOLFSSL_CERT_MANAGER* cm, WOLFSSL* ssl, int ret,
     if (ret == 0) {
         verify_ok = 1;
     }
-
+#if defined(WOLFSSL_ALT_CERT_CHAINS) && defined(OPENSSL_COMPATIBLE_DEFAULTS)
+    /* OpenSSL sets verification as "ok" if the peer certificate validate to a trust CA
+     * when the "alternate" cerfificate chain mode is used. Therefore, the result
+     * is overwritten by "ok" when it is CA, and it is passed to callback.
+     */
+    if (!verify_ok && ret == ASN_NO_SIGNER_E && args->dCert->isCA) {
```

Comment:
In my opinion this is not enough information to clear the error. Even with `OPENSSL_COMPATIBLE_DEFAULTS`.